src/pal/src/loader/module.cpp

```diff
@@ -676,7 +676,7 @@ PAL_RegisterModule(
 {
     HINSTANCE hinstance = nullptr;
 
-    int err = PAL_InitializeDLL();
+    int err = PAL_InitializeDLLWithoutSEH();
```

Comment:
PAL_RegisterModule/UnregisterModule can still be removed as long as the "local" module handle is passed to the DllMain. By "local" I mean the PAL calling the DllMain for LoadLibrary instead of calling PAL_RegisterModule. The reason is that sos, dbi and dac all share the same PAL (in the dac) so it will work. dbgshim doesn't have a DllMain handler so it is ok. I just don't know about VS.